cmd/kubeadm/app/cmd/upgrade/common_test.go

```diff
@@ -23,6 +23,136 @@ import (
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 )
 
+func TestGetK8sVersionFromUserInput(t *testing.T) {
+	var tcases = []struct {
+		name               string
+		isVersionMandatory bool
+		flags              *applyPlanFlags
+		args               []string
+		expectedErr        bool
+		expectedVersion    string
+	}{
+		{
+			name:               "No config and version as an argument",
+			isVersionMandatory: true,
+			flags:              &applyPlanFlags{},
+			args:               []string{"v1.13.1"},
+			expectedVersion:    "v1.13.1",
+		},
+		{
+			name:               "Neither config nor version specified",
+			isVersionMandatory: true,
+			flags:              &applyPlanFlags{},
+			expectedErr:        true,
+		},
+		{
+			name:               "No config and empty version as an argument",
+			isVersionMandatory: true,
+			flags:              &applyPlanFlags{},
+			args:               []string{""},
+			expectedErr:        true,
+		},
+		{
+			name:               "Valid config, but no version specified",
+			isVersionMandatory: true,
+			flags: &applyPlanFlags{
+				cfgPath: "testdata/cluster_config_with_version.yaml",
+			},
+			expectedVersion: "v1.13.0",
+		},
+		{
+			name:               "Valid config and different version specified",
+			isVersionMandatory: true,
+			flags: &applyPlanFlags{
+				cfgPath: "testdata/cluster_config_with_version.yaml",
+			},
+			args:            []string{"v1.13.1"},
+			expectedVersion: "v1.13.1",
+		},
+		{
+			name:               "No version in config and no version in arguments",
+			isVersionMandatory: true,
+			flags: &applyPlanFlags{
+				cfgPath: "testdata/cluster_config_without_version.yaml",
+			},
+			// XXX: This is possibly wrong. The current logic is that if version is omitted
+			//      from both the config and arguments then take the latest version
+			//      from the release server. Which is unknown at test time.
+			expectedVersion: "v1.13.3",
```

Comment:
This is a flaky test, it requires internet connection and changing every time we release a new version. Please, don't add this case.

Reply:
I added this test case to illustrate a possible problem. I wasn't sure If it's expected that a user could use `apply` without providing any version and still get away with it.

Removed now.